### PR TITLE
docs: CSI mount_options are available only for filesystem vols

### PR DESCRIPTION
**Pull request by tgross**
Partial fix for https://github.com/hashicorp/nomad/issues/9021

The CSI specifcation allows only the `file-system` attachment mode to have
mount options. The `block-device` mode is left "intentionally empty, for now"
in the protocol. We should be validating against this problem, but our
documentation also had it backwards.

Also adds missing docs for `mount_options` on the group's `volume` spec.

Fixing the validation is in https://github.com/hashicorp/nomad/pull/9044

### CODE REVIEW

**Comment by tgross** [comment]
Converting this back to draft while we hash out some of these issues in https://github.com/hashicorp/nomad/issues/9021#issuecomment-705106863, because this might not be the right approach.

File: website/pages/docs/commands/volume/register.mdx [hunk 1]
```diff
@@ -82,14 +82,16 @@ context {
...
-  `"file-system"`, to mount pre-formatted file system volumes. Some
-  storage providers will support `"block-device"`, which will require
-  the job be configured with appropriate mount options.
-
-- `mount_options` <code>([mount_options][]:nil)</code> - Options for
-  mounting `block-device`volumes without a pre-formatted file system.
+- `attachment_mode` `(string: <required>)` - The storage API that will be used
+  by the volume. Most storage providers will support `"file-system"`, to mount
+  volumes using the CSI filesystem API. Some storage providers will support
+  `"block-device"`, which mount the volume with the CSI block device API
```
Comment (cgbaker) [inline website/pages/docs/commands/volume/register.mdx:88]:
```suggestion
  `"block-device"`, which will mount the volume with the CSI block device API
```

File: website/pages/docs/commands/volume/register.mdx [hunk 2]
```diff
@@ -82,14 +82,16 @@ context {
...
+  within the container.
+
+- `mount_options` <code>([mount_options][]:nil)</code> - Options for mounting
```
Comment (cgbaker) [inline website/pages/docs/commands/volume/register.mdx:91]: not sure what the `[mount_options]` bit inside the parentheses is supposed to convey... it occurs below as well.

**Review by cgbaker — APPROVED** [review]
question about mount_options; otherwise, preview checks out.

File: website/pages/docs/commands/volume/register.mdx (hunk 2 above)
Comment (tgross) [reply website/pages/docs/commands/volume/register.mdx:91]: Good catch... that's in the existing docs but it doesn't render as any kind of link. Will fix.

**Review by notnoop — APPROVED** [review]

**Pull request merged**